### ISSUE TITLE
creatable_fields & updatable_fields for primary Id

### ISSUE THREAD

**Issue by Ehsan-Irshad** [JSONAPI-Resources/jsonapi-resources#673]
Hi,

I am new to ruby and specially to this gem. After reading on the code and examples I figure out that if I want certain fields (ID in my case) to be excluded from JSON request I can use the following code:

```
class ApiResource < JSONAPI::Resource

  def self.updatable_fields(context)
    super - [:id]
  end

  def self.creatable_fields(context)
    super - [:id]
  end
end
```

`

I inherit all my other resources from this class. So I dont accept the ID attribute for any of the JSON call for POST method.

But its not working and ID is always passed to the ActiveRecord to be save in DB.
How can I achieve this? Or is my knowledge wrong on the api.

My POST Request from POSTMAN:

```
`{"data":
    {
    "id":"5",
    "type":"applications",
    "attributes": {
        "name":"Ehsan",
        "clientAppId":"Client"
        }
    }
}
```

**Comment by barelyknown** [comment]
Is that the response or your request?

**Comment by Ehsan-Irshad** [comment]
Thats a request, data for POST

**Comment by Ehsan-Irshad** [comment]
anyone have an idea? Or we cant remove id?

**Comment by lgebhardt** [comment]
@Ehsan-Irshad The `id` shouldn't be getting updated on the model, and I don't see that it is. This gem does not update the database directly, but rather goes through the model.

In your post example the 'id' is used to identify the resource being updated, as per the [spec](http://jsonapi.org/format/#crud-updating), and is required. But again this gem does not update the `id` on the model, even though it is required to be in the request payload.

**Comment by Ehsan-Irshad** [comment]
@lgebhardt 
I think you are referring to PUT request where we update the model against a given specific ID?

Let me rephrase:
We use POST method to create a new model object. But for that model object we want ID to by null even if its passed in the request. If I save the model without the ID, DB will auto-generate (Auto Increment ID) for me. Thats the functionality I need.

**Comment by lgebhardt** [comment]
@Ehsan-Irshad Sorry, missed that it was a POST. I think #707 should fix it.

**Comment by Ehsan-Irshad** [comment]
@lgebhardt  Thank you.
I think I shall get the update once you merge it to master?
Using "gem update" command

**Comment by lgebhardt** [comment]
@Ehsan-Irshad You can test it by specifying the git repo and the branch in your Gemfile. 

``` ruby
gem 'jsonapi-resources', git: 'https://github.com/cerebris/jsonapi-resources.git', branch: 'id_respect_creatable'
```

You will also get the other changes that are in master, so a few other things may change.
